app/models/casa_case.rb

```diff
@@ -66,6 +66,22 @@ class CasaCase < ApplicationRecord
   delegate :name, to: :hearing_type, prefix: true, allow_nil: true
   delegate :name, to: :judge, prefix: true, allow_nil: true
 
+  # Validation class for :court_report_status & :court_report_submitted_at. Adds specific errors based on record data.
+  class CourtReportValidator < ActiveModel::Validator
+    def validate(record)
+      # Check for no timestamp and a submission value other than 'not_submitted'.
+      if record.court_report_submitted_at.nil? && record.court_report_status != "not_submitted"
+        record.errors[:court_report_status] << "Court report submission date can't be nil if status is anything but not_submitted."
+      # Check for timestamp with a 'not_submitted' status.
+      elsif !record.court_report_submitted_at.nil? && record.court_report_status == "not_submitted"
```

Comment:
I think it's also easier to not read double-negatives:

```
elsif record.court_report_submitted_at && record.court_report_not_submitted?
```

Since the `court_report_submitted_at` is a timestamp, it can only be `nil` or have a value, so it's sufficient to just check for `record.court_report_submitted_at`.